axlearn/common/checkpointer.py

```diff
@@ -922,3 +971,216 @@ def validate_and_restore(*, step: int, ckpt_dir: str):
             restored_state = state
 
         return step, restored_state
+
+
+class OrbaxCheckpointer(BaseCheckpointer):
+    """A checkpointer that uses orbax CheckpointManager.
+
+    NOTE: While this class uses index files to do additional validation on checkpoint state, the
+    index file is not used for committing checkpoints (i.e., it is not reliable to check for the
+    presence of 'index' file). Instead, use `checkpoint_paths` to identify committed checkpoints.
+    """
+
+    # Disable lazy imports within the class, to avoid requiring orbax dependency globally.
```

Comment:
What is the concern with doing this? Orbax doesn't loop in that many additional dependencies.

Reply:
Mainly to avoid assuming that all users have orbax installed -- once this is stable I can move orbax into the core deps. (It's likely already implied by some other packages, so maybe this is less of an issue.)